idaes/generic_models/unit_models/distillation/reboiler.py

```diff
@@ -545,40 +560,107 @@ def rule_bottoms_enth(self, t):
+                                               outlvl=outlvl,
+                                               hold_state=True)
+
+        if not self.control_volume.properties_out[0].temperature.fixed:
+
+            # Estimating a good guess for the reboiler temperature based on
+            # a vapor fraction of 0.5, bubble and dew
+            # point. This helps with initializing the system within the two
+            # phase envelope before activating the boilup ratio constraint.
+
+            alpha = 0.5 * (
+                self.control_volume.properties_in[0].temperature_dew.value -
+                self.control_volume.properties_in[0].
+                temperature_bubble.value) + \
+                self.control_volume.properties_in[0].temperature_bubble.value
+
+            self.eq_boilup_ratio.deactivate()
+            self.control_volume.properties_out[0].temperature.fix(alpha)
+
+            if degrees_of_freedom(self) == 0:
+                with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                    res = solver.solve(self, tee=slc.tee)
+                init_log.info(
+                    "Initialization Step 1 Complete, {}.".
+                    format(idaeslog.condition(res))
+                )
+            else:
+                raise Exception("State vars fixed but degrees of freedom "
+                                "for reboiler is not zero during "
+                                "initialization.")
 
-        if solver is not None:
-            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
-                res = solver.solve(self, tee=slc.tee)
-            init_log.info(
-                "Initialization Complete, {}.".format(idaeslog.condition(res))
-            )
+            self.control_volume.properties_out[0].temperature.unfix()
+            self.eq_boilup_ratio.activate()
         else:
-            init_log.warning(
-                "Solver not provided during initialization, proceeding"
-                " with deafult solver in idaes.")
-            solver = get_default_solver()
+            # In this case, outlet temperature has already been fixed by user
+            # and boil up ratio is assumed not to be fixed, otherwise the
+            # system is over specified.
+
+            self.eq_boilup_ratio.deactivate()
+
+            if degrees_of_freedom(self) == 0:
+                with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                    res = solver.solve(self, tee=slc.tee)
+                init_log.info(
+                    "Initialization Step 1 Complete, {}.".
+                    format(idaeslog.condition(res))
+                )
+            else:
+                raise Exception("State vars fixed but degrees of freedom "
+                                "for reboiler is not zero during "
+                                "initialization.")
+
+            self.eq_boilup_ratio.activate()
+
+        if degrees_of_freedom(self) == 0:
```

Comment:
Do you need to check degrees of freedom again? Once we've got this far, shouldn't we know already? This might mean storing the DoF (or a boolean) earlier to use here.

Reply:
Agreed. Removing this but updated the error message to better explain what is expected to be fixed.